### PR TITLE
Autogluon versions

### PR DESCRIPTION
**Pull request by yinweisu**
*Issue #, if available:*

*Description of changes:*
Added `show_versions()` to provide system info, autogluon versions, and autogluon's direct dependency versions. This will be useful for user to provide bug report. 
A following PR will be created for the Bug Report Template

The output looks like this:
```bash
INSTALLED VERSIONS
------------------
date                 : 2022-02-24
time                 : 18:15:51.950408
python               : 3.7.12.final.0
OS                   : Linux
OS-release           : 5.4.0-1057-aws
Version              : #60~18.04.1-Ubuntu SMP Thu Sep 9 20:38:09 UTC 2021
machine              : x86_64
processor            : x86_64
num_cores            : 48
cpu_ram_mb           : 191102
cuda version         : 11.450.142.00
num_gpus             : 4
gpu_ram_mb           : [15109, 15109, 15109, 15109]
avail_disk_size_mb   : 726178

autogluon.common     : 0.3.2b20220118
autogluon.core       : 0.3.2b20220118
autogluon.features   : 0.3.2b20220118
autogluon.forecasting: 0.3.2b20220118
autogluon.tabular    : 0.3.2b20220118
autogluon.text       : 0.3.2b20220118
autogluon.vision     : 0.3.2b20220118
autogluon_contrib_nlp: 0.0.1
boto3                : 1.17.74
catboost             : 1.0.3
dask                 : 2021.11.2
distributed          : 2021.11.2
fairscale            : 0.4.5
fastai               : 2.5.3
gluoncv              : 0.11.0
gluonts              : 0.8.1
lightgbm             : 3.3.1
matplotlib           : 3.5.1
networkx             : 2.6.3
nptyping             : 1.4.4
numpy                : 1.20.3
omegaconf            : 2.1.1
pandas               : 1.3.5
PIL                  : 8.3.2
psutil               : 5.8.0
pytest               : 6.2.5
pytorch_lightning    : 1.5.10
ray                  : 1.7.0
requests             : 2.25.1
scipy                : 1.6.3
sentencepiece        : None
skimage              : 0.19.2
sklearn              : 1.0.2
smart_open           : 5.2.1
timm                 : 0.5.4
torch                : 1.10.1+cu102
torchmetrics         : 0.7.2
tqdm                 : 4.62.3
transformers         : 4.16.2
xgboost              : 1.4.2
```

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by szha** [comment]
Job PR-1563-1 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-1563/1/index.html

**Comment by Innixma** [comment]
This is awesome!

Consider adding:

1. Total System Memory
2. Total System CPUs

Stretch Goals:

1. Total System GPUs (that are detected)
2. CUDA Install version
3. Total System Disk space

**Comment by sxjscience** [comment]
You may refer to https://github.com/dmlc/gluon-nlp/blob/5d4bc9eba7226ea9f9aabbbd39e3b1e886547e48/scripts/benchmarks/benchmark_utils.py#L1039 also.

This file also includes code about measuring CPU/GPU memory.

File: core/src/autogluon/core/utils/utils.py [hunk 1]
```diff
@@ -885,3 +896,9 @@ def unevaluated_fi_df_template(features: List[str]) -> pd.DataFrame:
         'n': 0
     }, index=features)
     return importance_df
+
+
+def bytes_to_mega_bytes(memory_amount: int) -> int:
+    """ Utility to convert a number of bytes (int) into a number of mega bytes (int)
+    """
+    return memory_amount >> 20
```
Comment (sxjscience) [inline core/src/autogluon/core/utils/utils.py:904]: new line.

**Review by sxjscience — APPROVED** [review]
LGTM overall. Just one formatting issue (missing new line).

**Comment by szha** [comment]
Job PR-1563-2 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-1563/2/index.html

**Pull request merged**